File: C++/260. Single Number III.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& n) {
        vector<int>a;
        map<int,int>m;
        for(int i:n)
        m[i]++;
        for(auto i=m.begin();i!=m.end();i++){
            if(i->second==1)
            a.push_back(i->first);
        }
        return a;
    }
};

File: JAVA/260. Single Number III.java
class Solution {
    public int[] singleNumber(int[] n) {
        List<Integer> a = new ArrayList<>();
        Map<Integer, Integer> m = new HashMap<>();
        for (int i : n) m.put(i, m.getOrDefault(i, 0) + 1);
        for (Map.Entry<Integer, Integer> i : m.entrySet()) {
            if (i.getValue() == 1) a.add(i.getKey());
        }
        return a.stream().mapToInt(Integer::intValue).toArray();
    }
}
